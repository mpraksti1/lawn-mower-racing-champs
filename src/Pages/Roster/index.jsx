import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import PlayerList from '../../Components/PlayerList';
import Button from '../../Elements/Button';
import Text from '../../Elements/Text';

const headerNames = ['first_name', 'last_name', 'rating', 'handedness'];

const Roster = ({
  retrievePlayers, deletePlayer, players,
}) => (
  <div className="roster-page">
    <Text xlg sans block spaceAbove>Your LMRC Roster:</Text>
    <Text sm thin spaceBelow block>{'Below are the players you\'ve picked so far'}</Text>
    <Button outline spaceBelow>
      <Link to="/player/new">
        <Text sm green>Add New</Text>
      </Link>
    </Button>
    <PlayerList
      headerNames={headerNames}
      retrievePlayers={retrievePlayers}
      deletePlayer={deletePlayer}
      players={players}
    />
  </div>
);

Roster.propTypes = {
  retrievePlayers: PropTypes.func.isRequired,
  deletePlayer: PropTypes.func.isRequired,
  players: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default Roster;
