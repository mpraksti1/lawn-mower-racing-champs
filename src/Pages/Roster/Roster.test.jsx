import Roster from '.';

describe('Roster page component', () => {
  it('renders without crashing', () => {
    shallow(<Roster retrievePlayers={() => {}} deletePlayer={() => {}} players={[]} />);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(
      <Roster retrievePlayers={() => {}} deletePlayer={() => {}} players={[]} />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
