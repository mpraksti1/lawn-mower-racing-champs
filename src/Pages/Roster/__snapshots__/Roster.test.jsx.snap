// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Roster page component matches snapshot 1`] = `
<div
  className="roster-page"
>
  <Text
    block={true}
    sans={true}
    spaceAbove={true}
    xlg={true}
  >
    Your LMRC Roster:
  </Text>
  <Text
    block={true}
    sm={true}
    spaceBelow={true}
    thin={true}
  >
    Below are the players you've picked so far
  </Text>
  <Button
    outline={true}
    spaceBelow={true}
  >
    <Link
      replace={false}
      to="/player/new"
    >
      <Text
        green={true}
        sm={true}
      >
        Add New
      </Text>
    </Link>
  </Button>
  <WithAlert(PlayerList)
    deletePlayer={[Function]}
    headerNames={
      Array [
        "first_name",
        "last_name",
        "rating",
        "handedness",
      ]
    }
    players={Array []}
    retrievePlayers={[Function]}
  />
</div>
`;
